### PR TITLE
feat: provide way to average profiles in different repetitions

### PR DESCRIPTION
**Pull request by JoanFM**

### CODE REVIEW

**Review by maateen — CHANGES_REQUESTED** [review]
Please resolve all merge conflicts.

File: src/utils/benchmark.py [hunk 1]
```diff
@@ -31,19 +34,35 @@ def benchmark_time(
...
             teardown()
 
         results.append(t.duration)
 
+    mean_profiles = []
+    for profile_cls, profile_list in profiles_by_cls.items():
+        mean_profiles.append(merge_profiles(profile_list))
+
     m = mean(results)
     s = stdev(results) if len(results) > 1 else None
```
Comment (davidbp) [inline src/utils/benchmark.py:64]: If there is no deviation I would put simply 0

Comment (JoanFM) [reply src/utils/benchmark.py:64]: We are putting None already, I feel it is more correct, but we can change it in another PR

**Review by maateen — APPROVED** [review]
LGTM

**Pull request merged**